### DISCUSSION TITLE
IBMZ support for push gateway

### DISCUSSION THREAD

**Discussion by modassarrana89** [prometheus/pushgateway#536] [General]
I am trying IBMZ support on pushgateway . I do have made changes on dockerignore , makefile. But i have few doubts that need certain clarification

1. In Makefile.common , its trying to use prefix as pwd for storing pushgateway binary . But in dockerfile , its expects it to be placed under ./build/{os}-{arch}/.  how is this directory getting created . In Makefile.common , it doesn't create this folder
```
Makefile.common
...
PREFIX                  ?= $(shell pwd)
...
.PHONY: common-build
common-build: promu
        @echo ">> building binaries"
        $(PROMU) build --prefix $(PREFIX) $(PROMU_BINARIES)

Dockerfile
...
COPY --chown=nobody:nobody .build/${OS}-${ARCH}/pushgateway /bin/pushgateway
...
```
Makefile.common do invokes dockerfile to build the image for pushgateway , but since it doesn't able to find the  .build directory , it is currently failing

To make it work , i am right now moving pushgateway binary stored in my present working directory to .build/linux-s390x/
& then it does work.

Please guide in which step of makefile.common or how does this build directory gets created 

3. In circleCI , i can see it test with different arch , but didn't find where its getting passed . I do want it to run for s390x. i have found that circleci do support self hosted s390x runner

**Comment by beorn7** [comment]
Yeah, I had the same problem, and it's the same for all Prometheus repos (as they all use the same Makefile.common). I think it's some `promu` magic that manages those directories. (Maybe it's called as `promu crossbuild` or something? Wildly guessing…) We need somebody with more promu foo.

Since this problem is not specific to the prometheus/pushgateway repo, it might be more likely to reach someone with the right qualifications if posting on the [Prometheus developers mailing list](https://groups.google.com/forum/#!forum/prometheus-developers).
